src/test/java/org/opensearch/security/configuration/ConfigurationRepositoryTest.java

```diff
@@ -133,10 +198,77 @@ public void initOnNodeStart_withSecurityIndexCreationEnabledShouldSetInstallDefa
+        when(clusterBlocks.hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)).thenReturn(false);
+
+        configRepository.initOnNodeStart().join();
+
+        verify(clusterBlocks, times(1)).hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE);
+    }
+
+    @Test
+    public void initOnNodeStart_whenClusterInitiallyUnavailable() {
+        Settings settings = Settings.builder().put(SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true).build();
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+        when(clusterBlocks.hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)).thenReturn(true, true, false);
+
+        configRepository.initOnNodeStart().join();
+
+        verify(clusterBlocks, times(3)).hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE);
+    }
+
+    @Test
+    public void initOnNodeStart_installDefaultConfigSuccess() throws InterruptedException, TimeoutException {
+        // Required for ComplianceConfig in audit.yml to work
+        setupObjectMapperInjectables();
+
+        Settings settings = Settings.builder()
+            .put(SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true)
+            .put("path.home", TEST_CONFIG_DIR)
+            .put(ConfigConstants.SECURITY_UNSUPPORTED_DELAY_INITIALIZATION_SECONDS, 1)
+            .build();
+        System.setProperty("security.default_init.dir", TEST_CONFIG_DIR);
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+        configRepository.setDynamicConfigFactory(dynamicConfigFactory);
+
+        setupConfigurationLoaderMock(configurationLoaderSecurity7, CType.values());
+
+        // Cluster Status mocks
+        when(clusterBlocks.hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)).thenReturn(false);
+
+        // Node mocks
+        when(discoveryNode.getName()).thenReturn("node1");
+
+        CreateIndexResponse createIndexResponse = new CreateIndexResponse(true, true, ".opendistro_security");
+
+        ClusterHealthResponse clusterHealthResponse = mock(ClusterHealthResponse.class);
+
+        IndexResponse indexResponse = mock(IndexResponse.class);
+
+        when(indicesAdminClient.create(any(CreateIndexRequest.class))).thenReturn(
+            getCreateIndexResponsePlainActionFuture(createIndexResponse)
+        );
+        when(clusterAdminClient.health(any(ClusterHealthRequest.class))).thenReturn(
+            getClusterHealthResponsePlainActionFuture(clusterHealthResponse)
+        );
+        when(localClient.index(any(IndexRequest.class))).thenReturn(getIndexResponsePlainActionFuture(indexResponse));
+
+        setupIndexResponseForDefaultConfig(indexResponse);
+
+        when(dynamicConfigFactory.isInitialized()).thenReturn(false, true);
+
+        configRepository.initOnNodeStart().join();
+
+        verify(clusterBlocks, times(1)).hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE);
+        verify(indexResponse, times(10)).getId();
```

Comment:
What is special about the number 10 here? Is this the number of CType?

Reply:
yes, I can change it to ``Ctypes.values.size()``